### PR TITLE
ci: replace circleci shellchecker with gha

### PR DESCRIPTION
**Pull request by michael-siek**
## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [ ] Code is reviewed for security

### CODE REVIEW

File: .github/workflows/shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: shellcheck
+
+# Only run tests on unprotected branches.
+# Otherwise we waste build minutes re-running tests will know will pass.
+on:
+  push:
+    branches:
+      - '**'
+      - '!develop'
+      - '!main'
```
Comment (stephenmathieson) [inline .github/workflows/shellcheck.yml]: This is a public repository, so minutes are free.

```suggestion
on: [push]
```

**Review by stephenmathieson — DISMISSED** [review]
Small thing.

Is this repo still in use by anything?

**Comment by michael-siek** [comment]
https://github.com/dequelabs/axe-core-npm/blob/develop/.circleci/config.yml#L257-L268

**Review by stephenmathieson — APPROVED** [review]
We should be migrating to GHA rather than worrying about this 🤷

**Pull request merged**